openpype/plugins/publish/integrate_new.py

```diff
@@ -110,801 +156,494 @@ class IntegrateAssetNew(pyblish.api.InstancePlugin):
-            self.log.warning(msg)
+            # Manage anatomy template data
+            template_data.pop("frame", None)
+            if repre.get("udim"):
+                template_data["udim"] = repre["udim"][0]
 
-        return filled_template
+            # Construct destination filepath from template
+            anatomy_filled = anatomy.format(template_data)
+            template_filled = anatomy_filled[template_name]["path"]
+            repre_context = template_filled.used_values
+            dst = os.path.normpath(template_filled)
 
-    def create_version(self, subset, version_number, data=None):
-        """ Copy given source to destination
+            # Single file transfer
+            src = os.path.join(stagingdir, fname)
+            transfers = [(src, dst)]
 
-        Args:
-            subset (dict): the registered subset of the asset
-            version_number (int): the version number
+        for key in self.db_representation_context_keys:
+            value = template_data.get(key)
+            if not value:
+                continue
+            repre_context[key] = template_data[key]
 
-        Returns:
-            dict: collection of data to create a version
-        """
+        # Explicitly store the full list even though template data might
+        # have a different value
+        if repre.get("udim"):
+            repre_context["udim"] = repre.get("udim")  # store list
 
-        return {"schema": "openpype:version-3.0",
-                "type": "version",
-                "parent": subset["_id"],
-                "name": version_number,
-                "data": data}
+        # Use previous representation's id if there is a name match
+        existing = existing_repres_by_name.get(repre["name"].lower())
+        if existing:
+            repre_id = existing["_id"]
+        else:
+            repre_id = ObjectId()
+
+        # Backwards compatibility:
+        # Store first transferred destination as published path data
+        # todo: can we remove this?
+        published_path = transfers[0][1]
+        repre["published_path"] = published_path  # Backwards compatibility
+
+        # todo: `repre` is not the actual `representation` entity
+        #       we should simplify/clarify difference between data above
+        #       and the actual representation entity for the database
+        data = repre.get("data", {})
+        data.update({'path': published_path, 'template': template})
+        representation = {
+            "_id": repre_id,
+            "schema": "openpype:representation-2.0",
+            "type": "representation",
+            "parent": version["_id"],
+            "name": repre['name'],
+            "data": data,
+            "dependencies": instance.data.get("dependencies", "").split(),
+
+            # Imprint shortcut to context for performance reasons.
+            "context": repre_context
+        }
+
+        # todo: simplify/streamline which additional data makes its way into
```

Comment:
This is directly relate to work on this https://github.com/pypeclub/OpenPype/pull/1572 which unfortunately got stalled, but we considered it a prerequisite before moving onto a major integration refactor. It is mean to solve this exact issue and make representation a lot more defined.

Reply:
> It is mean to solve this exact issue and make representation a lot more defined.

Super. I'll look into it!